source/fuzz/fuzzer.h

```diff
@@ -72,6 +88,38 @@ class Fuzzer {
       protobufs::TransformationSequence* transformation_sequence_out) const;
 
  private:
+  // A convenience method to add a repeated fuzzer pass to |pass_instances| with
+  // probability 0.5.
+  //
+  // All fuzzer passes take |ir_context|, |transformation_context|,
+  // |fuzzer_context| and |transformation_sequence_out| as parameters.  Extra
+  // arguments can be provided via |extra_args|.
+  template <typename FuzzerPassT, typename... Args>
+  void MaybeAddRepeatedPass(
+      RepeatedPassInstances* pass_instances, opt::IRContext* ir_context,
+      TransformationContext* transformation_context,
+      FuzzerContext* fuzzer_context,
+      protobufs::TransformationSequence* transformation_sequence_out,
+      Args&&... extra_args) const;
+
+  // A convenience method to add a final fuzzer pass to |passes| with
+  // probability 0.5.
+  //
+  // All fuzzer passes take |ir_context|, |transformation_context|,
+  // |fuzzer_context| and |transformation_sequence_out| as parameters.  Extra
+  // arguments can be provided via |extra_args|.
+  template <typename FuzzerPassT, typename... Args>
+  void MaybeAddFinalPass(
+      std::vector<std::unique_ptr<FuzzerPass>>* passes,
+      opt::IRContext* ir_context, TransformationContext* transformation_context,
+      FuzzerContext* fuzzer_context,
+      protobufs::TransformationSequence* transformation_sequence_out,
+      Args&&... extra_args) const;
+
+  bool ContinueFuzzing(
```

Comment:
Nit: I find the name a bit misleading, since this is only deciding whether the fuzzer should continue fuzzing, rather than actually continuing.

Could something similar to ShouldContinueFuzzing be better?

Reply:
I added the comment and changed the name - agreed on both fronts.